java/common/src/main/scala/org/typelevel/otel4s/java/IOLocalContextStorage.scala

```diff
@@ -0,0 +1,36 @@
+package org.typelevel.otel4s.java
+
+import cats.effect.IOLocal
+import cats.effect.LiftIO
+import cats.effect.std.Dispatcher
+import io.opentelemetry.context.Context
+import io.opentelemetry.context.ContextStorage
+import io.opentelemetry.context.Scope
+
+class IOLocalContextStorage[F[_]: LiftIO](
+    dispatcher: Dispatcher[F],
+    ioLocal: IOLocal[Context]
+) extends ContextStorage {
+
+  override def attach(toAttach: Context): Scope =
+    dispatcher.unsafeRunSync(
+      currentOrRoot
+        .flatMap { old =>
+          ioLocal
+            .set(toAttach)
+            .as(new Scope {
+              def close() = dispatcher.unsafeRunSync(ioLocal.set(old).to[F])
+            })
+        }
+        .to[F]
```

Comment:
It very nicely attaches in this IOLocal... and is invisible to `current()`, which runs on a different fiber.

Reply:
Trying to get around this with `IOLocals`.